lib/perl/Genome/Model/DeNovoAssembly/Command/ExportAsGscProject.pm

```diff
@@ -0,0 +1,176 @@
+
+    my @inst_data = $model->instrument_data;
+    return if $assembler =~ /newbler/i && @inst_data > 1;
+    return 1;
+    #return if ! exists $subjects_to_copy{ $model->subject->name };
+}
+
+sub subdirs_for_assembler {
+    my ($self, $assembler) = validate_pos(@_, {isa => __PACKAGE__}, {type => SCALAR});
+    return if not $supported_assemblers{$_[1]};
+    @{$supported_assemblers{$_[1]}->{subdirs}};
+}
+
+sub assemblers_edit_dir {
+    my ($self, $assembler) = validate_pos(@_, {isa => __PACKAGE__}, {type => SCALAR});
+    return if not $supported_assemblers{$_[1]};
+    $supported_assemblers{$_[1]}->{edit_dir};
+}
+
+sub additional_files_for_assembler {
+    my ($self, $build) = validate_pos(@_, {isa => __PACKAGE__}, {isa => 'Genome::Model::Build'});
+
+    my @addl_files;
+    if( $build->model->processing_profile->assembler_name eq 'newbler de-novo-assemble' ) {
+        for my $file ( glob( File::Spec->join($build->data_directory, "*input.fastq")) ) {
+            push @addl_files, [ $file, 'edit_dir' ];
+        }
+        push @addl_files, [ File::Spec->join($build->data_directory, 'consed', 'phdball_dir', 'phd.ball.1'), File::Spec->join('phdball_dir') ];
+    }
+
+    return @addl_files;
+}
+
+sub execute {
+    my $self = shift;
+
+    my $succeeded_builds = $self->_resolve_builds;
+    for my $build ( @$succeeded_builds ) {
+        $self->status_message("Export: %s %s", $build->model->__display_name__, $build->data_directory);
+        $self->_export_build($build);
+    }
+
+    return 1;
+}
+
+sub _resolve_builds {
+    my $self = shift;
+
+    my $project = $self->project;
+    $self->status_message('Resolving builds for project: %s', $project->__display_name__);
+
+    my @project_parts = $project->parts(entity_class_name => 'Genome::Model::DeNovoAssembly');
+    $self->fatal_message("No de novo models associated with %s", $project->__display_name__) unless @project_parts;
+    $self->status_message('Associated Models: %s', scalar(@project_parts));
+
+    my @models = map { $_->entity } @project_parts;
+    $self->fatal_message("Models associated with %s do not exist!", $project->__display_name__) unless @models;
+    $self->status_message('Existing Models: %s ', scalar(@models));
+
+    my @succeeded_builds;
+    for my $model ( @models ) {
+        next if ! $self->is_model_supported($model);
+        push @succeeded_builds, $model->last_succeeded_build;
+    }
+
+    $self->status_message('Succeeded builds: %s', scalar(@succeeded_builds));
+    $self->fatal_message('No succeeded builds found!') if not @succeeded_builds;
+
+    return \@succeeded_builds;
+}
+
+sub _export_build {
+    my ($self, $build) = @_;
+
+    my $build_dir = $build->data_directory;
+    $self->fatal_message("Build data directory does not exist! $build_dir") unless -d $build_dir;
+
+    # project dir
+    my $copy_dir = $self->directory;
+    my $output_dir = File::Spec->join($copy_dir, $build->model->subject->name);
+    $output_dir =~ s/\s+/_/g;
```

Comment:
This seems like a job for `Genome::Utility::Text`'s `sanitize_string_for_filesystem`!  (I'd be tempted to include the ID in the path to avoid collisions, though.)